repair_stock_move/models/stock_move.py

```diff
@@ -62,6 +66,6 @@ def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
         )
         if self.repair_line_id and self.repair_line_id.lot_id:
             vals["lot_id"] = self.repair_line_id.lot_id.id
-        elif self.repair_id and self.repair_id.lot_id:
+        elif self.repair_id and not self.repair_line_id and self.repair_id.lot_id:
```

Comment:
Could this change affect the lots in the lines?

Reply:
Moves for repair lines have both `repair_id` and `repair_line_id` filled

https://github.com/OCA/repair/blob/5a872c57b52bd150ad208b97babff884e4cc6824/repair_stock_move/models/repair_line.py#L26-L27

and move for repair header only has `repair_id`:

https://github.com/OCA/repair/blob/5a872c57b52bd150ad208b97babff884e4cc6824/repair_stock_move/models/repair_order.py#L61-L73

So added `elif` will only affect to repair header move, not for repair lines move.

Reply:
I see it now: "if self.repair_line_id and self.repair_line_id.lot_id:" does the thing for the lines.